logpuzzle.py

```diff
@@ -41,13 +48,35 @@ def download_images(img_urls, dest_dir):
     Creates the directory if necessary.
     """
     # +++your code here+++
-    pass
+    if not os.path.exists(dest_dir) or not os.path.isdir(dest_dir):
+        os.mkdir(dest_dir)
+    else:
+        print('directory exists')
+
+    img_files = []
+    for img in sorted(img_urls,
+                      key=lambda w: re.search(r'\w*\.jpg', w).group()):
+        img_filename = re.search(r'[^/]*\.jpg', img).group()
+        img_files.append(img_filename)
+        if os.path.exists(os.path.join(dest_dir, img_filename)):
+            pass
+        else:
+            with contextlib.closing(urllib.urlopen(img)) as req:
+                with open(os.path.join(dest_dir, img_filename), 'w') as ifile:
+                    print('Saving {} to {}'.format(img, img_filename))
+                    ifile.write(req.read())
+
+    with open(os.path.join(dest_dir, 'index.html'), 'w') as html:
+        html.write('<html>\n<body style="display:flex">\n\t\t<img src="' +
+                   '"/>\n\t\t<img src="'.join(img_files) +
+                   '"/>\n\t</body></html>"')
```

Comment:
Add exception handling for network errors. The guidance section of the assessment for socket timeoutes, but HTTP 404 is also suggested.